libandroid-navigation-ui/src/main/java/com/mapbox/services/android/navigation/ui/v5/map/NavigationMapboxMap.java

```diff
@@ -59,13 +60,21 @@
   private static final int[] ZERO_MAP_PADDING = {0, 0, 0, 0};
   private static final double NAVIGATION_MAXIMUM_MAP_ZOOM = 18d;
   private final List<Marker> mapMarkers = new ArrayList<>();
+  private final CopyOnWriteArrayList<OnWayNameChangedListener> onWayNameChangedListeners
+    = new CopyOnWriteArrayList<>();
+  private final MapWayNameChangedListener internalWayNameChangedListener
+    = new MapWayNameChangedListener(onWayNameChangedListeners);
+  private final NavigationMapSettings settings = new NavigationMapSettings();
```

Comment:
Shouldn't this be an immutable class that is instantiated when it's needed?

Reply:
Added some explanation: https://github.com/mapbox/mapbox-navigation-android/pull/1700#discussion_r251827497

Reply:
@danesfeder what about making this not final?